idl/download.x

```diff
@@ -0,0 +1,41 @@
+%#include "block.x"
+
+namespace mazzaroth
+{
+    // Download Requests are made between Mazzaroth nodes to sync data
+    struct DownloadRequest {
+        DownloadRequestPayload downloadRequestPayload;
+    }
+
+    // The different request messages that can be sent
+    enum DownloadRequestType {
+        UNKNOWN = 0,
+        HEIGHT = 1, // Request current block height of node
+        BLOCK = 2 // Request a specific block from the node
+    };
+
+    union DownloadRequestPayload switch (DownloadRequestType Type)
+    {
+        case UNKNOWN:
+            void;
+        case HEIGHT:
```

Comment:
Requesting the block height seems like a more general operation rather than one specific to downloading blocks.

Probably doesn't need to be changed now, but may be out of place long term.